<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogManageRoutes</class>
 <widget class="QDialog" name="DialogManageRoutes">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Routes</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Common</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QWidget" name="widget" native="true">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="4" column="1">
           <widget class="QComboBox" name="route_prof"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_3">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Resolve domains to IP before connect, also affects the server address as well&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Outbound Domain Strategy</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QComboBox" name="outbound_domain_strategy">
            <property name="editable">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label">
            <property name="toolTip">
             <string>Sniffs the connections intial packet(s?) to identify the protocol and the SNI. Sniff result for destination also rewrites the destionation of the connection to the sniffed SNI. For more data on the sniffed protocol refer to the sing-box's documentation.</string>
            </property>
            <property name="text">
             <string>Sniffing Mode</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="toolTip">
             <string notr="true">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enable to resolve domains to IP before routing based on the strategy&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Inbound Domain Strategy</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="sniffing_mode">
            <item>
             <property name="text">
              <string>Disable</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Sniff result for routing</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="route_prof_l">
            <property name="text">
             <string>Routing Profile</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="domainStrategyCombo">
            <property name="editable">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="ruleset_mirror_l">
            <property name="text">
             <string>Remote Rule-set Mirror</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QComboBox" name="ruleset_mirror">
            <item>
             <property name="text">
              <string>GitHub</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>jsDelivr(Cloudflare)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>jsDelivr(Gcore)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>jsDelivr(Quantil)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>jsDelivr(Fastly)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>jsDelivr(CDN)</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Hijack</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QGroupBox" name="hijack_box">
         <property name="title">
          <string>DNS Server</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QCheckBox" name="dnshijack_enable">
            <property name="text">
             <string>Enable</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="dnshijack_what">
            <property name="text">
             <string>How does it work?</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="dnshijack_listenport_l">
            <property name="text">
             <string>Listen Port</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="dnshijack_listenport"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="dnshijack_v4resp_l">
            <property name="text">
             <string>IPv4 Response</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="dnshijack_v4resp"/>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="dnshijack_v6resp_l">
            <property name="text">
             <string>IPv6 Response</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLineEdit" name="dnshijack_v6resp"/>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="dnshijack_rules_l">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Following format should be used:&lt;br/&gt;domain:a.com&lt;/p&gt;&lt;p&gt;suffix:a.com&lt;/p&gt;&lt;p&gt;regex:someregex&lt;/p&gt;&lt;p&gt;rule_set:nameOfRuleSet&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Rules</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QTextEdit" name="dnshijack_rules"/>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="dnshijack_allow_lan">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Whether to listen on 0.0.0.0 so that other devices on LAN can use this server&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Allow Lan to Connect</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Redirect Settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="2" column="1">
           <widget class="QLineEdit" name="redirect_listenport"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="redirect_listenport_l">
            <property name="text">
             <string>Listen Port</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QCheckBox" name="redirect_enable">
            <property name="text">
             <string>Enable</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="redirect_listenaddr"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="redirect_listenaddr_l">
            <property name="text">
             <string>Listen Address</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>DNS</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="simple_dns_box">
         <property name="title">
          <string>Simple DNS Settings</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="0">
           <widget class="QLabel" name="label_9">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;examples:&lt;br/&gt;tls://8.8.8.8&lt;br/&gt;https://domain/path&lt;/p&gt;&lt;p&gt;tcp://8.8.8.8:1234&lt;/p&gt;&lt;p&gt;dhcp://auto&lt;/p&gt;&lt;p&gt;h3://domain/path&lt;/p&gt;&lt;p&gt;quic://domain:4632&lt;/p&gt;&lt;p&gt;etc&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Remote DNS</string>
            </property>
            <property name="buddy">
             <cstring>widget</cstring>
            </property>
           </widget>
          </item>
          <item row="5" column="2">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="dns_final_out_l">
              <property name="text">
               <string>Default DNS server</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="dns_final_out">
              <item>
               <property name="text">
                <string notr="true">remote</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">direct</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="2">
           <widget class="QLineEdit" name="local_override"/>
          </item>
          <item row="1" column="2">
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QComboBox" name="remote_dns">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="editable">
               <bool>true</bool>
              </property>
              <item>
               <property name="text">
                <string notr="true">tls://8.8.8.8</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">tls://1.1.1.1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">8.8.8.8</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">1.1.1.1</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Query Strategy</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="remote_dns_strategy"/>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_2">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Local DNS can not be used in some configurations, you will have to set an ip here to be used as local dns in such situations&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Local Override</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_8">
            <property name="toolTip">
             <string>This is especially important and it is recommended to use the default value of &quot;localhost&quot;.
If the default value does not work, try changing it to &quot;223.5.5.5&quot;.
For more information, see the document &quot;Configuration/DNS&quot;.</string>
            </property>
            <property name="text">
             <string>Direct DNS</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QComboBox" name="direct_dns">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="editable">
               <bool>true</bool>
              </property>
              <item>
               <property name="text">
                <string notr="true">localhost</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">223.5.5.5</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">119.29.29.29</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">178.22.122.100</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">77.88.8.8</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Query Strategy</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="direct_dns_strategy"/>
            </item>
           </layout>
          </item>
          <item row="5" column="0">
           <widget class="QCheckBox" name="enable_fakeip">
            <property name="text">
             <string>Enable FakeIP</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>DNS Object Settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QCheckBox" name="use_dns_object">
              <property name="text">
               <string>Use DNS Object</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="format_dns_object">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Format</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="dns_document">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Document</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="dns_object"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Route</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="route_profiles_box">
         <property name="title">
          <string>Routing Profiles</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QListWidget" name="route_profiles"/>
          </item>
          <item>
           <widget class="QWidget" name="widget_2" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QPushButton" name="new_route">
               <property name="text">
                <string>New</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="clone_route">
               <property name="text">
                <string>Clone</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="export_route">
               <property name="text">
                <string>Export</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="edit_route">
               <property name="text">
                <string>Edit</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="delete_route">
               <property name="text">
                <string>Delete</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>sniffing_mode</tabstop>
  <tabstop>domainStrategyCombo</tabstop>
  <tabstop>outbound_domain_strategy</tabstop>
  <tabstop>remote_dns</tabstop>
  <tabstop>remote_dns_strategy</tabstop>
  <tabstop>direct_dns</tabstop>
  <tabstop>direct_dns_strategy</tabstop>
  <tabstop>dns_final_out</tabstop>
  <tabstop>use_dns_object</tabstop>
  <tabstop>format_dns_object</tabstop>
  <tabstop>dns_document</tabstop>
  <tabstop>dns_object</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogManageRoutes</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogManageRoutes</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
